### PR TITLE
feat: upgrade designer-pytorch-2.3 related images to python 3.9

### PR DESCRIPTION
**Pull request by RealKai42**
upgrade designer-pytorch-2.3-train and designer-pytorch-2.3 to python 3.9

The reason we use pip==20.2 is that it provides the legacy resolver, which is necessary for installing our package.

### CODE REVIEW

File: assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml [hunk 1]
```diff
@@ -24,4 +23,5 @@ dependencies:
       - pycryptodome==3.19.1
       - pycryptodomex==3.19.1
       - gunicorn==22.0.0
-      - idna==3.7
\ No newline at end of file
+      - idna==3.7
+      - azureml-defaults==1.58.0
```
[the remark this answers is not in the document]
Comment (RealKai42) [reply assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml:27]: you mean like: `azureml-defaults=={{latest}}`  or `{{template("azureml-defaults")}}`

File: assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml [hunk 2]
```diff
@@ -5,17 +5,16 @@ channels:
...
-      - azureml-designer-pytorch-modules==0.0.69
+      - azureml-designer-pytorch-modules==0.0.71
       - future==0.18.3
       - numpy==1.22
-      - cryptography==42.0.4
       - azureml-dataprep==4.10.1
-      - Werkzeug==3.0.3
-      - pip==21.1
+      - Werkzeug==3.0.6
+      - pip==23.3
```
Comment (vizhur) [inline assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml:17]: pip should be defined as conda dependency, just update version above line#7

File: assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vizhur) [reply assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml:27]: azureml-defaults=={{latest-pypi-version}}

File: assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml (hunk 2 above)
Comment (RealKai42) [reply assets/designer/environments/designer-pytorch-2.3-train/context/conda_dependencies.yaml:17]: just like the comments in #3625, We need pip 20.2, which uses the legacy resolver, to install our package.

**Review by vizhur — APPROVED** [review]

**Pull request merged**